code/modules/reagents/reagent_containers/patch.dm

```diff
@@ -17,7 +17,7 @@
 	return // thanks inheritance again
 
 /obj/item/weapon/reagent_containers/pill/patch/canconsume(mob/eater, mob/user)
-	if(!iscarbon(eater))
+	if(!iscarbon(eater) || eater.stat == DEAD)
 		return 0
 	return 1 // Masks were stopping people from "eating" patches. Thanks, inheritance.
```

Comment:
all this "thanks inheritance" makes me think the person who made this can't code properly. oh wait, goofball coded it. that explains that, then.